File: app/macos/NativeStream/NativeStream/Views/Player/PlayerSidebarRow.swift
// PlayerSidebarRow.swift — UX-021
// Single row in the player's On Now sidebar tab.

import SwiftUI

struct PlayerSidebarRow: View {

    @Environment(EPGViewModel.self)    private var epgVM
    @Environment(PlayerViewModel.self) private var playerVM

    let channel: Channel

    private var isPlaying: Bool { playerVM.currentChannel?.id == channel.id }
    private var current: Programme? { epgVM.currentProgramme(for: channel) }
    private var next: Programme?    { epgVM.nextProgramme(for: channel) }
    private var isLive: Bool        { current != nil }

    var body: some View {
        Button {
            playerVM.play(channel: channel)
        } label: {
            HStack(spacing: NS.Spacing.sm) {

                ChannelLogoSquare(
                    channel: channel,
                    size: 32,
                    cornerRadius: NS.Radius.sm
                )

                VStack(alignment: .leading, spacing: 2) {
                    Text(channel.name)
                        .font(NS.Font.captionMed)
                        .foregroundStyle(Color.white.opacity(isPlaying ? 0.9 : 0.6))
                        .lineLimit(1)

                    if let prog = current ?? next {
                        Text(prog.title)
                            .font(NS.Font.monoSm)
                            .foregroundStyle(Color.white.opacity(0.3))
                            .lineLimit(1)
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)

                // Right indicator
                if isPlaying {
                    Image(systemName: "play.fill")
                        .font(.system(size: 9))
                        .foregroundStyle(NS.accent)
                } else if isLive {
                    NSPulseDot()
                } else if let next {
                    Text(next.startTimeString)
                        .font(NS.Font.monoSm)
                        .foregroundStyle(NS.accent.opacity(0.7))
                }
            }
            .padding(.horizontal, NS.Spacing.sm)
            .padding(.vertical, NS.Spacing.xs)
            .background(isPlaying ? NS.accentGlow : Color.clear)
            .clipShape(RoundedRectangle(cornerRadius: NS.Radius.md))
        }
        .buttonStyle(.plain)
    }
}

File: app/android/app/src/main/java/com/nativestream/android/ui/screens/player/PlayerSidebarRow.kt
// app/src/main/java/com/nativestream/android/ui/screens/player/PlayerSidebarRow.kt
//
// Single row in the On Now tab. Shows logo, channel name, programme title,
// and right indicator (▶ playing / time remaining / next start time).

package com.nativestream.android.ui.screens.player

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.nativestream.android.domain.model.Channel
import com.nativestream.android.ui.theme.NSColors
import com.nativestream.android.ui.theme.NSDimens
import com.nativestream.android.ui.theme.NSType
import com.nativestream.android.ui.viewmodel.EpgViewModel
import com.nativestream.android.ui.viewmodel.PlayerViewModel

private val PLAY_ICON_SIZE = 9.dp

@Composable
fun PlayerSidebarRow(
    channel: Channel,
    playerViewModel: PlayerViewModel,
    epgViewModel: EpgViewModel,
    modifier: Modifier = Modifier,
) {
    val dimens        = NSDimens.current
    val activeChannel by playerViewModel.activeChannel.collectAsState()

    val isPlaying = activeChannel?.id == channel.id
    val current   = epgViewModel.currentProgramme(channel)
    val next      = epgViewModel.nextProgramme(channel)
    val programme = current ?: next

    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(dimens.spacing.sm),
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(dimens.radius.md))
            .background(if (isPlaying) NSColors.accentGlow else Color.Transparent)
            .clickable { playerViewModel.play(channel) }
            .padding(horizontal = dimens.spacing.sm, vertical = dimens.spacing.xs),
    ) {
        ChannelLogoSquare(
            channel      = channel,
            size         = dimens.channel.logoSquareSmall,
            cornerRadius = dimens.radius.sm,
        )

        Column(
            verticalArrangement = Arrangement.spacedBy(2.dp),
            modifier = Modifier.weight(1f),
        ) {
            Text(
                text     = channel.name,
                style    = NSType.captionMedium(),
                color    = Color.White.copy(alpha = if (isPlaying) 0.9f else 0.6f),
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            programme?.let {
                Text(
                    text     = it.title,
                    style    = NSType.monoSmall(),
                    color    = Color.White.copy(alpha = 0.3f),
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }
        }

        // Right indicator
        when {
            isPlaying -> Icon(
                imageVector        = Icons.Default.PlayArrow,
                contentDescription = "Now playing",
                tint               = NSColors.accent,
                modifier           = Modifier.size(PLAY_ICON_SIZE),
            )
            current != null -> Text(
                text  = current.timeRemainingString,
                style = NSType.monoSmall(),
                color = Color.White.copy(alpha = 0.3f),
            )
            next != null -> Text(
                text  = next.startTimeString,
                style = NSType.monoSmall(),
                color = NSColors.accent.copy(alpha = 0.7f),
            )
        }
    }
}